stability: span used after call to Finish()

```
panic: span used after call to Finish():
&{tracer:<nil> event:0xd9a140 Mutex:{state:0 sema:0} raw:{Context:{TraceID:6383740812092313299 SpanID:320797998694638280
3 Sampled:false Baggage:map[]} ParentSpanID:0 Operation:sql txn Start:{sec:63617893721 nsec:285667018 loc:0x2ae8660} Dur
ation:538694040 Tags:map[] Logs:[]} numDroppedLogs:0}

goroutine 1122428 [running]:
panic(0x1880780, 0xc431ec7220)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).maybeAssertSanityLocked(0xc42d
bfe6e0)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.go:31 +0x341
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).Lock(0xc42dbfe6e0)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.go:25 +0x3f
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).LogFields(0xc42dbfe6e0, 0xc431
fbe300, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/span.go:123 +0xd3
github.com/cockroachdb/cockroach/pkg/util/tracing.(*TeeSpan).LogFields(0xc42fae6280, 0xc431fbe300, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tee_tracer.go:183 +0x73
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x7fc1bd6a69d0, 0xc42d00ef40, 0x100, 0x1a37b25, 0x15, 0xc428
2b9e28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:132 +0x4bc
github.com/cockroachdb/cockroach/pkg/util/log.VEventf(0x7fc1bd6a69d0, 0xc42d00ef40, 0xc400000002, 0x1a37b25, 0x15, 0xc42
82b9e28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:203 +0x107
github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1.1(0xc427923590, 0xc42ad4d540, 0x7fc1bd6a69d0, 0x
c42d00ef40, 0xc420440f60, 0xc420440f70, 0xc420440f88)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:188 +0x14c
github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1(0xc427923590, 0xc42ad4d540, 0x7fc1bd6a69d0, 0xc4
2d00ef40, 0xc42f8a03c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:198 +0x80
created by github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:201 +0x101
```

Cc @cockroachdb/stability, @spencerkimball

The same thing happened on cyan last night as well:

```
E161221 06:01:26.456970 18492 server.go:2315  http: TLS handshake error from 13.64.168.11:4768: EOF
panic: span used after call to Finish():
&{tracer:<nil> event:0xd9a1e0 Mutex:{state:0 sema:0} raw:{Context:{TraceID:1152939163692707338 SpanI
D:8504425252134737952 Sampled:false Baggage:map[]} ParentSpanID:0 Operation:sql txn Start:{sec:63617
896885 nsec:394849398 loc:0x2ae8660} Duration:1847112888 Tags:map[] Logs:[]} numDroppedLogs:0}

goroutine 19183 [running]:
panic(0x1880880, 0xc425648d00)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).maybeAsser
tSanityLocked(0xc421a61c20)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.
go:31 +0x341
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).Lock(0xc42
1a61c20)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.
go:25 +0x3f
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).LogFields(
0xc421a61c20, 0xc4241b5a80, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/span.g
o:123 +0xd3
github.com/cockroachdb/cockroach/pkg/util/tracing.(*TeeSpan).LogFields(0xc421a4f280, 0xc4241b5a80, 0
x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tee_tracer.go:183 +0x73
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x7f05b13b23f8, 0xc427011580, 0x100, 0x1
a37c25, 0x15, 0xc421e7fe28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:132 +0x4bc
github.com/cockroachdb/cockroach/pkg/util/log.VEventf(0x7f05b13b23f8, 0xc427011580, 0xc400000002, 0x1a37c25, 0x15, 0xc421e7fe28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:203 +0x107
github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1.1(0xc42745da90, 0xc421d0fcc0, 0x7f05b13b23f8, 0xc427011580, 0xc420018250, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:188 +0x14c
github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1(0xc42745da90, 0xc421d0fcc0, 0x7f05b13b23f8, 0xc427011580, 0xc420e96720)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:198 +0x80
created by github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:201 +0x101
```

--- a/pkg/kv/transport.go
+++ b/pkg/kv/transport.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/envutil"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/opentracing/opentracing-go"
 )
 
@@ -163,7 +164,11 @@ func (gt *grpcTransport) SendNext(ctx context.Context, done chan<- BatchCall) {
 	gt.clientIndex++
 	gt.setPending(client.args.Replica, true)
 
+	// Fork the original context as this async send may outlast the
+	// caller's context.
+	ctx, sp := tracing.ForkCtxSpan(ctx, "grpcTransport SendNext")
 	go func() {
+		defer tracing.FinishSpan(sp)
 		gt.opts.metrics.SentCount.Inc(1)
 		reply, err := func() (*roachpb.BatchResponse, error) {
 			if enableLocalCalls {

